setuptools/tests/test_virtualenv.py

```diff
@@ -43,11 +43,11 @@ def bare_virtualenv():
 SOURCE_DIR = os.path.join(os.path.dirname(__file__), '../..')
 
 
-def test_clean_env_install(bare_virtualenv):
+def test_clean_env_install(bare_virtualenv, tmp_src):
     """
     Check setuptools can be installed in a clean environment.
     """
-    bare_virtualenv.run(['python', 'setup.py', 'install'], cd=SOURCE_DIR)
+    bare_virtualenv.run(['python', 'setup.py', 'install'], cd=tmp_src)
```

Comment:
In my environment, `tmp_src` causes a copy of over 200MB of data (including the whole git repo). Perhaps there's a more selective way to isolate the operations, such as by pointing the 'build' dir to a temporary directory.

Reply:
After further exploration, I don't believe it's currently possible to cause the 'sdist' command not to create a directory in the same directory as the project root, so copying is necessary.

Reply:
I can imagine doing something like `git worktree` that doesn't copy `.git/` dir over but rather maintains a link to the main repo location.